etcdserver/server.go

```diff
@@ -553,11 +553,21 @@ func (s *EtcdServer) run() {
 		<-appdonec
 	}()
 
-	select {
-	case err := <-s.errorc:
-		plog.Errorf("%s", err)
-		plog.Infof("the data-dir used by this member must be removed.")
-	case <-s.stop:
+	for {
+		select {
+		case leases := <-s.lessor.ExpiredLeasesC():
+			go func() {
+				for _, l := range leases {
+					s.LeaseRevoke(context.TODO(), &pb.LeaseRevokeRequest{ID: int64(l.ID)})
```

Comment:
I think multi revoke should be all-or-nothing-- may be difficult to implement correctly. Is there a major advantage to batching lease revokes?